cli/azd/extensions/microsoft.ai.agents/internal/cmd/context.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/azure/azure-dev/cli/azd/pkg/azdext"
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+)
+
+func newContextCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "context",
+		Short: "Get the context of the AZD project & environment.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Create a new context that includes the AZD access token
+			ctx := azdext.WithAccessToken(cmd.Context())
+
+			// Create a new AZD client
+			azdClient, err := azdext.NewAzdClient()
+			if err != nil {
+				return fmt.Errorf("failed to create azd client: %w", err)
+			}
+
+			defer azdClient.Close()
+
+			hasEnv := false
+
+			getConfigResponse, err := azdClient.UserConfig().Get(ctx, &azdext.GetUserConfigRequest{
+				Path: "",
+			})
+			if err == nil {
+				if getConfigResponse.Found {
+					color.HiWhite("User Config")
+					var userConfig map[string]string
+					err := json.Unmarshal(getConfigResponse.Value, &userConfig)
+					if err == nil {
+						jsonBytes, err := json.MarshalIndent(userConfig, "", "  ")
+						if err == nil {
+							fmt.Println(string(jsonBytes))
+						}
+					}
+				}
+			}
+
+			getProjectResponse, err := azdClient.Project().Get(ctx, &azdext.EmptyRequest{})
+			if err == nil {
+				color.Cyan("Project:")
+
+				projectValues := map[string]string{
+					"Name": getProjectResponse.Project.Name,
+					"Path": getProjectResponse.Project.Path,
+				}
+
+				for key, value := range projectValues {
+					fmt.Printf("%s: %s\n", color.HiWhiteString(key), value)
+				}
+				fmt.Println()
+			} else {
+				color.Yellow("WARNING: No azd project found in current working directory")
+				fmt.Printf("Run %s to create a new project.\n", color.CyanString("azd init"))
+				return nil
+			}
+
+			var currentEnvName string
+
+			getEnvResponse, err := azdClient.Environment().GetCurrent(ctx, &azdext.EmptyRequest{})
+			if err == nil {
+				currentEnvName = getEnvResponse.Environment.Name
+				hasEnv = true
+			} else {
+				color.Yellow("WARNING: No azd environment(s) found.")
+				fmt.Printf("Run %s to create a new environment.\n", color.CyanString("azd env new"))
+				return nil
+			}
+
+			environments := []string{}
```

Comment:
Very minor NIT. If you're just going to append to it anyways you can just give it a nil slice and it works. I think there's a linter warning in some linters for this:

```suggestion
   var environments []string
```